Python/sysmodule.c

```diff
@@ -3358,6 +3358,25 @@ make_impl_info(PyObject *version_info)
     if (res < 0)
         goto error;
 
+#ifdef _WIN32
+# define OS_PLATFORM "win32"
+#elif _WIN64
+# define OS_PLATFORM "amd64"
+#elif _M_ARM
+# define OS_PLATFORM "arm32"
+#elif _M_ARM64
+# define OS_PLATFORM "arm64"
```

Comment:
We don't want to hard code the list in source - we want to provide a preprocessor variable in the `pythoncore.vcxproj` file set to `$(ArchName)`. Then here we check if that's defined, and if it is, then create a string from it and add it to the impl_info dict. If it's not defined, we can skip it.

Have a look in `pythoncore.vcxproj` for `VPATH` and `$(PyVpath)`, and search for that in the referenced files to see how it's used. That one should be a better example than any of the others we currently have.

Reply:
Thanks for the guidance, I will try to migrate the preprocessing into `pythoncore.vcxproj`.